documents/invoice/test_inv.go

```diff
@@ -0,0 +1,97 @@
+// +build integration unit testworld
+
+package invoice
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	coredocumentpb "github.com/centrifuge/centrifuge-protobufs/gen/go/coredocument"
+	"github.com/centrifuge/go-centrifuge/contextutil"
+	"github.com/centrifuge/go-centrifuge/documents"
+	"github.com/centrifuge/go-centrifuge/identity"
+	testingidentity "github.com/centrifuge/go-centrifuge/testingutils/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+func (b *Bootstrapper) TestBootstrap(context map[string]interface{}) error {
+	return b.Bootstrap(context)
+}
+
+func (*Bootstrapper) TestTearDown() error {
+	return nil
+}
+
+func invData(t *testing.T) []byte {
+	did, err := identity.NewDIDFromString("0xEA939D5C0494b072c51565b191eE59B5D34fbf79")
+	assert.NoError(t, err)
+	dec, err := documents.NewDecimal("42")
+	assert.NoError(t, err)
+	data := Data{
+		Recipient: &did,
+		Currency:  "EUR",
+		LineItems: []*LineItem{
+			{
+				ItemNumber:  "12312431",
+				Description: "just",
+				Quantity:    dec,
+			},
+		},
+	}
+
+	d, err := json.Marshal(data)
+	assert.NoError(t, err)
+	return d
+}
+
+func CreateInvPayload(t *testing.T, collaborators []identity.DID) documents.CreatePayload {
+	if collaborators == nil {
+		collaborators = []identity.DID{testingidentity.GenerateRandomDID()}
+	}
+	return documents.CreatePayload{
+		Scheme: Scheme,
+		Collaborators: documents.CollaboratorsAccess{
+			ReadWriteCollaborators: collaborators,
+		},
+		Data: invData(t),
+	}
+}
+
+func InitInvoice(t *testing.T, did identity.DID, payload documents.CreatePayload) *Invoice {
+	inv := new(Invoice)
+	assert.NoError(t, inv.unpackFromCreatePayload(did, payload))
+	return inv
+}
+
+func CreateInvWithEmbedCDWithPayload(t *testing.T, ctx context.Context, did identity.DID, payload documents.CreatePayload) (*Invoice, coredocumentpb.CoreDocument) {
```

Comment:
we seem to have different names for invoice in functions - `Inv` and `Invoice`. Let's be consistent?

Reply:
I would recommend `Invoice`